From: Kei Tokunaga <ktokunag@redhat.com>
Subject: [RHEL5.1 PATCH 17/21] xm save/restore does not work on IA64
Date: Thu, 07 Jun 2007 03:46:15 -0400
Bugzilla: 240858
Message-Id: <4667B7C7.9020108@redhat.com>
Changelog: [xen] ia64: xm save/restore does not work


bz240858
https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=240858

Backport of cset#14831.

At RHEL5.1, xm save/restore related function does not
work correctly on IA64.  Time interpolator remembers
the last status.  When a domain is restored, it should
be flushed.  Otherwise the interpolator generates a bad
timer value.

Thanks,
Kei


rh bug 240858

# HG changeset patch
# User Alex Williamson <alex.williamson@hp.com>
# Date 1176353170 21600
# Node ID e4bd31a66a2e2bc70faa3851cb8706d791b38f99
# Parent  3d356a2b1c75c2fea9b8eb0643075614e9e3d4fe
[IA64] Bugzilla 942 Fix (timer strangeness on restored PV domain)

Time interpolator remembers last status.  When domain is restored, it
should be forgotten. Otherwise the interpolator causes bad timer value.

Signed-off-by: Isaku Yamahata <yamahata@valinux.co.jp>


---

 linux-2.6.18-21.el5-gerd-order-kei/arch/ia64/xen/hypervisor.c |    3 +++
 1 file changed, 3 insertions(+)

diff -puN arch/ia64/xen/hypervisor.c~14831-IA64_Bugzilla_942_Fix_timer_strangeness_on_restored_PV_domain arch/ia64/xen/hypervisor.c
--- linux-2.6.18-21.el5-gerd-order/arch/ia64/xen/hypervisor.c~14831-IA64_Bugzilla_942_Fix_timer_strangeness_on_restored_PV_domain	2007-06-07 02:44:33.000000000 -0400
+++ linux-2.6.18-21.el5-gerd-order-kei/arch/ia64/xen/hypervisor.c	2007-06-07 02:44:33.000000000 -0400
@@ -858,6 +858,9 @@ time_resume(void)
 
 	/* Just trigger a tick.  */
 	ia64_cpu_local_tick();
+
+	/* Time interpolator remembers the last timer status.  Forget it */
+	time_interpolator_reset();
 }
 
 ///////////////////////////////////////////////////////////////////////////

_

